### ISSUE TITLE
Retraining of Doc2Vec Model

### ISSUE THREAD

**Issue by charleywu** [piskvorky/gensim#263]
Please add the functionality to retrain a Doc2Vec model on new documents

**Comment by transcranial** [comment]
How would one go about doing this? Trying to recreate the IMDB dataset experimental protocol described in the original Le and Mikolov paper. From the paper:

`Experimental protocols: We learn the word vectors and paragraph vectors using 75,000 training documents (25,000 labeled and 50,000 unlabeled instances). The paragraph vectors for the 25,000 labeled instances are then fed through a neural network with one hidden layer with 50 units and a logistic classifier to learn to predict the sentiment. At test time, given a test sentence, we again freeze the rest of the network and learn the paragraph vectors for the test reviews by gradient descent. Once the vectors are learned, we feed them through the neural network to predict the sentiment of the reviews.`

Is there a straightforward way of learning the paragraph vectors for the test samples? Retraining seems to build the vocabulary again. Can one somehow append to the vocab list with words appearing in the test samples but not the training samples? Otherwise, I suppose to learn the test sample paragraph vectors one just uses the weights from the initially trained model as the starting point for further learning.
